solidity/random-beacon/test/RandomBeacon.Relay.test.ts

```diff
@@ -574,34 +663,92 @@ describe("RandomBeacon - Relay", () => {
           // `groupSize * relayEntrySubmissionEligibilityDelay +
           // relayEntryHardTimeout`.
           await mineBlocks(64 * 10 + 5760)
-
-          tx = await randomBeacon.connect(notifier).reportRelayEntryTimeout()
         })
 
-        it("should slash the full slashing amount for all group members", async () => {
-          await expect(tx)
-            .to.emit(staking, "Seized")
-            .withArgs(to1e18(1000), 5, notifier.address, membersAddresses)
+        context("when group creation is warmed up", () => {
+          beforeEach(async () => {
+            // Simulate group creation was warmed up upon request.
+            await (randomBeacon as RandomBeaconStub).publicCreateGroupWarmUp()
 
-          await expect(tx)
-            .to.emit(randomBeacon, "RelayEntryTimeoutSlashed")
-            .withArgs(1, to1e18(1000), membersAddresses)
-        })
+            tx = await randomBeacon.connect(notifier).reportRelayEntryTimeout()
+          })
 
-        it("should terminate the group", async () => {
-          // TODO: Implementation once `Groups` library is ready.
-        })
+          it("should slash the full slashing amount for all group members", async () => {
+            await expect(tx)
+              .to.emit(staking, "Seized")
+              .withArgs(to1e18(1000), 5, notifier.address, membersAddresses)
 
-        it("should emit RelayEntryTimedOut event", async () => {
-          await expect(tx)
-            .to.emit(randomBeacon, "RelayEntryTimedOut")
-            .withArgs(1, 0)
+            await expect(tx)
+              .to.emit(randomBeacon, "RelayEntryTimeoutSlashed")
+              .withArgs(1, to1e18(1000), membersAddresses)
+          })
+
+          it("should terminate the group", async () => {
+            // TODO: Implementation once `Groups` library is ready.
+          })
+
+          it("should emit RelayEntryTimedOut event", async () => {
+            await expect(tx)
+              .to.emit(randomBeacon, "RelayEntryTimedOut")
+              .withArgs(1, 0)
+          })
+
+          it("should clean up current relay request data", async () => {
+            // TODO: Uncomment those assertions once termination is implemented.
+            // await expect(tx).to.not.emit(randomBeacon, "RelayEntryRequested")
+            // expect(await randomBeacon.isRelayRequestInProgress()).to.be.false
+          })
+
+          it("should cool down group creation", async () => {
+            // TODO: Uncomment those assertions once termination is implemented.
+            // expect(await randomBeacon.getGroupCreationState()).to.be.equal(
+            //   dkgState.IDLE
+            // )
+            // expect(await sortitionPool.isLocked()).to.be.false
+          })
```

Comment:
To make it simpler and do not repeat checks that are not related to the warmup (awaiting seed) state, I'd do:

```
when no active groups exist after timeout is reported:
  ... put all the checks here
when no active groups exist after timeout is reported and DKG is awaiting seed:
  ... just assertions about: a) changing DKG state; b) unlocking pool
```

Reply:
And I believe assertions a) b) can be now implemented and do not require any uncomment TODO.

Reply:
See https://github.com/keep-network/keep-core/pull/2691/commits/1797b660235df348f267aabb9e9e929e07a88e15. However, assertions must be commented until we have group termination implemented. If the group is not terminated we enter in the `groups.numberOfActiveGroups() > 0` branch in `reportRelayEntryTimeout`. Once we have termination we can enter in the `else` and test the behavior of `notifySeedTimedOut`